src/compiler/checker.ts

```diff
@@ -6502,6 +6499,17 @@ namespace ts {
                 return result;
             }
 
+            function typeNodeIsEquivalentToType(typeNode: TypeNode, annotatedDeclaration: Declaration, type: Type) {
+                const typeFromTypeNode = getTypeFromTypeNode(typeNode);
+                if (typeFromTypeNode === type) {
+                    return true;
+                }
+                if ((isParameter(annotatedDeclaration) || !compilerOptions.exactOptionalPropertyTypes && isPropertyDeclaration(annotatedDeclaration)) && annotatedDeclaration.questionToken) {

```

Comment:
Maybe it’s undesirable to do this for property declarations even if `exactOptionalPropertyTypes` is disabled, because it would start emitting types that are more restrictive for users who are consuming them with `exactOptionalPropertyTypes`. I’ll revert this part. I’m also going to look into using the available type alias even when we can’t fully reuse the type node, so converting this to a draft.